packages/hooks/src/useStorageState/index.ts

```diff
@@ -25,7 +34,7 @@ function useStorageState<T>(storage: Storage, key: string, defaultValue?: T | IF
   function updateState(value?: T | IFuncUpdater<T>) {
     if (typeof value === 'undefined') {
       storage.removeItem(key);
-      setState(defaultValue);
+      setState(undefined);
```

Comment:
This will cause a breaking change. We should be careful about it... @brickspert

Reply:
Sure，this is a break change. I think here should not be changed.

After change , if i use `setState(undefined)`, the state is `undefined`. but when render the component next, the state will be defaultValue, it's incorrect.

Reply:
Hi guys, thanks for the review. 
I looked at documentation and tests but nowhere was specified what value would be returned after deletion of the key.

@brickspert

```javascript
const hook = setUp({ key: 'key', defaultValue: 'value' });
act(() => {
 hook.result.current.setState('value');
});
expect(hook.result.current.state).toEqual('value');
act(() => {
    hook.result.current.setState('newValue');
});
expect(hook.result.current.state).toEqual('newValue');
act(() => {
    hook.result.current.setState(undefined);
});
hook.rerender({ key: 'key', defaultValue: 'value1' });

/* Which value would you expect here? (undefined, 'value1' or 'value') */
expect(hook.result.current.state).toBeUndefined();
hook.rerender();
expect(hook.result.current.state).toBeUndefined();
```

Reply:
@jvince You are right. The document only has a piece of description about `setState(undefined)` :

> If you want to delete this record from localStorage, you can use `setState()` or `setState(undefined)`.

As you have said, what the new state would be after calling `setState(undefined)` is actually an **undefined behavior** now. For now, it resets the localStorage and change the state to the `defaultValue`. But for the `setState(undefined)` of React `useState`, the state will just change to be `undefined`. Frankly, I'm worried about this inconsistency. This is really a serious problem we have to think over. Thanks for your contribution!